File: tests/org.eclipse.rap.rwt.test/src/org/eclipse/rap/rwt/widgets/Composite_Test.java
/*******************************************************************************
 * Copyright (c) 2002-2006 Innoopract Informationssysteme GmbH.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 * 
 * Contributors:
 *     Innoopract Informationssysteme GmbH - initial API and implementation
 ******************************************************************************/

package org.eclipse.rap.rwt.widgets;

import junit.framework.TestCase;
import org.eclipse.rap.rwt.RWT;
import org.eclipse.rap.rwt.RWTFixture;
import com.w4t.engine.lifecycle.PhaseId;

public class Composite_Test extends TestCase {

  protected void setUp() throws Exception {
    RWTFixture.setUp();
    RWTFixture.fakePhase( PhaseId.PROCESS_ACTION );
  }

  protected void tearDown() throws Exception {
    RWTFixture.tearDown();
  }

  public void testTabList() {
    Display display = new Display();
    Composite shell = new Shell( display , RWT.NONE );
    // add different controls to the shell
    Control button = new Button( shell, RWT.PUSH );
    Control link = new Link( shell, RWT.NONE );
    new Label( shell, RWT.NONE );
    new Sash( shell, RWT.HORIZONTAL );
    Combo combo = new Combo( shell, RWT.DROP_DOWN );
    Composite composite = new Composite( shell, RWT.NONE );
    List list = new List( shell, RWT.NONE );
    Text text = new Text( shell, RWT.NONE );
    Control[] controls = shell.getTabList();
    // check that the right ones are in the list
    assertEquals( 6, controls.length );
    assertEquals( button, controls[ 0 ] );
    assertEquals( link, controls[ 1 ] );
    assertEquals( combo, controls[ 2 ] );
    assertEquals( composite, controls[ 3 ] );
    assertEquals( list, controls[ 4 ] );
    assertEquals( text, controls[ 5 ] );
  }

}

File: tests/org.eclipse.rap.rwt.q07.jstest/js/org/eclipse/rwt/test/tests/CompositeTest.js
/*******************************************************************************
 * Copyright (c) 2010 EclipseSource and others. All rights reserved.
 * This program and the accompanying materials are made available under the
 * terms of the Eclipse Public License v1.0 which accompanies this distribution,
 * and is available at http://www.eclipse.org/legal/epl-v10.html
 *
 * Contributors:
 *   EclipseSource - initial API and implementation
 ******************************************************************************/

qx.Class.define( "org.eclipse.rwt.test.tests.CompositeTest", {
  extend : qx.core.Object,
  
  members : {
    
    TARGETENGINE : [ "mshtml" ],
        
    testCompositeBackgroundInitial : function() {
      var testUtil = org.eclipse.rwt.test.fixture.TestUtil;
      var composite = new org.eclipse.swt.widgets.Composite();
      composite.setBackgroundColor( null );
      composite.setBackgroundImage( null );
      composite.addToDocument();
      testUtil.flush();
      var image = composite._getTargetNode().style.backgroundImage;
      assertTrue( image.indexOf( "blank.gif" ) != -1 );
      composite.destroy();
      testUtil.flush();
    },
        
    testCompositeBackgroundFromColor : function() {
      var testUtil = org.eclipse.rwt.test.fixture.TestUtil;
      var composite = new org.eclipse.swt.widgets.Composite();
      composite.setBackgroundColor( "green" );
      composite.setBackgroundImage( null );
      composite.addToDocument();
      testUtil.flush();
      var image = composite._getTargetNode().style.backgroundImage;
      assertEquals( "", image );
      composite.setBackgroundColor( null );
      image = composite._getTargetNode().style.backgroundImage;
      assertTrue( image.indexOf( "blank.gif" ) != -1 );
      composite.destroy();
      testUtil.flush();
    },
        
    testCompositeBackgroundFromImage : function() {
      var testUtil = org.eclipse.rwt.test.fixture.TestUtil;
      var composite = new org.eclipse.swt.widgets.Composite();
      composite.setBackgroundColor( null );
      composite.setBackgroundImage( "bla.jpg" );
      composite.addToDocument();
      testUtil.flush();
      var image = composite._getTargetNode().style.backgroundImage;
      assertEquals( "url(bla.jpg)", image );
      composite.setBackgroundImage( null );
      image = composite._getTargetNode().style.backgroundImage;
      assertTrue( image.indexOf( "blank.gif" ) != -1 );
      composite.destroy();
      testUtil.flush();
    }

  }
  
} );